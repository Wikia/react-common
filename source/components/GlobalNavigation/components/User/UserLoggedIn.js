import React from 'react';
import PropTypes from 'prop-types';

import Dropdown from '../../../Dropdown';
import Avatar from '../../../Avatar';
import List from '../../../List';

import LinkText from '../Link/LinkText';
import LinkLogOut from '../Link/LinkLogOut';

const UserToggle = ({ chevron, data }) => (
    <React.Fragment>
        <Avatar alt={data.username} src={data.avatar_url} />
        {chevron}
    </React.Fragment>
);

UserToggle.propTypes = {
    chevron: PropTypes.node.isRequired,
    data: PropTypes.object.isRequired,
};

/* eslint-disable react/no-array-index-key */
const UserLoggedIn = ({ data }) => (
    <Dropdown
        toggle={chevron => <UserToggle chevron={chevron} data={data} />}
        className="wds-global-navigation__user-menu wds-global-navigation__user-logged-in"
        contentScrollable={false}
        contentClassName="wds-global-navigation__dropdown-content"
        iconName="dropdown-tiny"
        hasShadow
        dropdownRightAligned
    >
        <List isLinked>
            {data.items.map((item, index) => {
                if (item.type === 'link-text') {
                    return <li key={index}><LinkText link={item} /></li>;
                }

                if (item.type === 'link-logout') {
                    return <li key={index}><LinkLogOut link={item} /></li>;
                }

                return null;
            })}
        </List>
    </Dropdown>
);

UserLoggedIn.propTypes = {
    data: PropTypes.object.isRequired,
};

export default UserLoggedIn;
