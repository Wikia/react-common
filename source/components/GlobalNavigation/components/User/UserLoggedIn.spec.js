import React from 'react';
import { shallow } from 'enzyme';
import merge from 'lodash/merge';

import UserLoggedIn from './UserLoggedIn';

const defaultProps = {
    data: {
        avatar_url: 'https://fandomdesignsystem.com/images/FANDOM-Avatar.jpg',
        username: 'Ryba777',
        'tracking-label': 'account',
        items: [
            {
                type: 'link-text',
                title: {
                    type: 'translatable-text',
                    key: 'global-navigation-user-view-profile',
                },
                href: '//starwars.wikia.com/wiki/User:Ryba777',
                'tracking-label': 'account.profile',
            },
            {
                type: 'link-text',
                title: {
                    type: 'translatable-text',
                    key: 'global-navigation-user-my-talk',
                },
                href: '//starwars.wikia.com/wiki/User_talk:Ryba777',
                'tracking-label': 'account.talk',
            },
            {
                type: 'link-text',
                title: {
                    type: 'translatable-text',
                    key: 'global-navigation-user-my-preferences',
                },
                href: '//starwars.wikia.com/wiki/Special:Preferences',
                'tracking-label': 'account.preferences',
            },
            {
                type: 'link-text',
                title: {
                    type: 'translatable-text',
                    key: 'global-navigation-user-help',
                },
                href: '//community.wikia.com/wiki/Help:Contents',
                'tracking-label': 'account.help',
            },
            {
                type: 'link-logout',
                href: 'https://www.wikia.com/logout',
                title: {
                    type: 'translatable-text',
                    key: 'global-navigation-user-sign-out',
                },
                'param-name': 'returnto',
                'tracking-label': 'account.sign-out',
            },
            {
                type: 'link-mock',
                href: 'https://www.wikia.com/mock',
                title: {
                    type: 'translatable-text',
                    key: 'global-navigation-user-sign-out',
                },
                'param-name': 'mock-param',
                'tracking-label': 'account.mock-out',
            },
        ],
    },
};

function renderComponent(props) {
    const computedProps = merge({}, defaultProps, props);

    return shallow(<UserLoggedIn {...computedProps} />);
}

test('UserLoggedIn renders correctly', () => {
    expect(renderComponent()).toMatchSnapshot();
});

test('UserLoggedIn renders toggle correctly', () => {
    const wrapper = renderComponent();
    const Toggle = wrapper.find('Dropdown').props().toggle(<div />);

    expect(shallow(Toggle)).toMatchSnapshot();
});
