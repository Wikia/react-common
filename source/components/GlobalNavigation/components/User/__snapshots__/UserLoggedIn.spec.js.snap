// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`UserLoggedIn renders correctly 1`] = `
<Dropdown
  canFlip={false}
  className="wds-global-navigation__user-menu wds-global-navigation__user-logged-in"
  contentClassName="wds-global-navigation__dropdown-content"
  contentScrollable={false}
  dropdownLeftAligned={false}
  dropdownRightAligned={true}
  hasDarkShadow={false}
  hasShadow={true}
  iconName="dropdown-tiny"
  isActive={false}
  isLevel2={false}
  isNotHoverable={false}
  isStickedToParent={false}
  noChevron={false}
  onClose={null}
  onMouseEnter={null}
  toggle={[Function]}
  toggleAttrs={Object {}}
  toggleClassName=""
>
  <List
    bigItems={false}
    boldItems={false}
    className=""
    hasEllipsis={false}
    isLinked={true}
    linesBetween={false}
  >
    <li
      key="0"
    >
      <LinkText
        isStandaloneLink={false}
        link={
          Object {
            "href": "//starwars.wikia.com/wiki/User:Ryba777",
            "title": Object {
              "key": "global-navigation-user-view-profile",
              "type": "translatable-text",
            },
            "tracking-label": "account.profile",
            "type": "link-text",
          }
        }
      />
    </li>
    <li
      key="1"
    >
      <LinkText
        isStandaloneLink={false}
        link={
          Object {
            "href": "//starwars.wikia.com/wiki/User_talk:Ryba777",
            "title": Object {
              "key": "global-navigation-user-my-talk",
              "type": "translatable-text",
            },
            "tracking-label": "account.talk",
            "type": "link-text",
          }
        }
      />
    </li>
    <li
      key="2"
    >
      <LinkText
        isStandaloneLink={false}
        link={
          Object {
            "href": "//starwars.wikia.com/wiki/Special:Preferences",
            "title": Object {
              "key": "global-navigation-user-my-preferences",
              "type": "translatable-text",
            },
            "tracking-label": "account.preferences",
            "type": "link-text",
          }
        }
      />
    </li>
    <li
      key="3"
    >
      <LinkText
        isStandaloneLink={false}
        link={
          Object {
            "href": "//community.wikia.com/wiki/Help:Contents",
            "title": Object {
              "key": "global-navigation-user-help",
              "type": "translatable-text",
            },
            "tracking-label": "account.help",
            "type": "link-text",
          }
        }
      />
    </li>
    <li
      key="4"
    >
      <ComponentWrapper
        link={
          Object {
            "href": "https://www.wikia.com/logout",
            "param-name": "returnto",
            "title": Object {
              "key": "global-navigation-user-sign-out",
              "type": "translatable-text",
            },
            "tracking-label": "account.sign-out",
            "type": "link-logout",
          }
        }
      />
    </li>
  </List>
</Dropdown>
`;

exports[`UserLoggedIn renders toggle correctly 1`] = `
<Fragment>
  <Avatar
    alt="Ryba777"
    src="https://fandomdesignsystem.com/images/FANDOM-Avatar.jpg"
  />
  <div />
</Fragment>
`;
